add 8 sampler and small sampler option to latenight

nopeppermint opened the pull request:
This add's the option to have 4 or 8 sampler's in skin latenight

![4samplers](https://user-images.githubusercontent.com/3403218/46906183-d2000880-ceff-11e8-8ffe-6e8fa55501cd.png)
![8samplers](https://user-images.githubusercontent.com/3403218/46906184-d4faf900-ceff-11e8-92a2-14bbc8e68dd8.png)

nopeppermint committed:
add 8 sampler to latenight, with option for small samplers

first draw of small sampler, remove no longer used files and minor fix

add expand/collapse button, bpm, tooltip and smaller font

4 small or big sampler, or 8 tiny sampler with either the 4 left or 4 right big

remove 4 sampler option, adapt small sampler for two rows

remove no longer used buttons, add new sampler idea

remove spacers and change attribute names

fix tab/whitespace

ronso0 commented:
While I've been fancying this for a while, too, I believe we also need minimal samplers like in the other skins.
Maybe we can put a expand/collapse button in between the left-hand & right-hand samplers and pick up the style of the respective button in the FX units?

ronso0 commented:
"Load/Save sampler bank" buttons can go to SAMPLERS section.
MISCELLANEOUS can become LIBRARY now.

nopeppermint commented:
> we also need minimal samplers like in the other skins.

that's my next step ;-)

> "Load/Save sampler bank" buttons can go to SAMPLERS section.
MISCELLANEOUS can become LIBRARY now.

I agree

nopeppermint commented:
![smal_and_big_sampler](https://user-images.githubusercontent.com/3403218/46907220-88b7b500-cf0f-11e8-90f5-7c3e818ebb82.png)

ronso0 commented:
> Most all-in-one style controllers have one or two rows of 4 sampler buttons on each side, which is why we have rows of 8 in the other skins.

This is only true for Tango & Deere, and 8 samplers only fit into Deere (at minimal size) because the hotcue buttons are missing. Squeezing 8 samplers per row into LateNight's current layout, and with its big buttons everywhere, would require a massive rework of the expanded sampler template.
I consider this PR a good first step to get more samplers at all.

Later on, we can think about working around the space limitations while considering the 'common' controller layout. For example we could have 8 minmal samplers per row, separated into 1-4 and 5-8, then expand each block of 4 samplers by switching back to a 2x2 grid.

ronso0 commented:
The expand button seems kinda lost in between the samplers blocks. I suppose it could expand to the sampler row's height. This way it would stay in the same place under the mouse cursor when expanding/collapsing a row.
Also, I feel the button's drop shadow only makes sense in brighter button containers (like decks etc). I suggest to omit it (the bg-image) here and use a gradient for the button border instead. I'd also try to replace the `+/-` icons with text labels like `1-4` and `5-8`. That would make the buttons wider but it would also pick up the descriptive FX expand buttons' style.

nopeppermint commented:
now there are more options:

if you have only 4 samplers (no changes here):
=> 4 small Samplers in a row
=>4 big Samplers in a row

if you have 8 samplers:
=> 8 tiny Samplers in a row

![8_sampler_per_row](https://user-images.githubusercontent.com/3403218/46980825-1c3ae280-d0d6-11e8-83cd-28465dd6a0e2.png)

=> either the Samplers 1-4 big in a row
=> or the Samplers 5-6 big in a row

TODO: 
=> change the "+/-" Button to "+/- 1-4" and  "+/- 5-8"
=> maybe change color of Sampler 5-8 to the same as deck 3/4 
(how can this be done easily? 
adding color: #09B2AE; somewhere..)

ronso0 commented:
I'm sorry to say the last commit feels like a regression. The toggling is intransparent, samplers are suddenly gone when expanding others.

What about making 8 samplers the standart? The foundation is already there: drop the complicating  '4/8 samplers' option and let them expand like this (IMO all samplers can be expanded, small samplers 5-8 are just for illustration):
![latenight-8samplers-alt](https://user-images.githubusercontent.com/5934199/47024808-b6486c80-d162-11e8-91d2-318046db1e94.png)


Furthermore, I think we can split the space in the small samplers (next to Play) into two rows: top row shows just the title, while the bottom rows host BPM, Eject, xFade orientation (and maybe other controls)

What do you think?

nopeppermint commented:
> let's drop the complicating '4/8 samplers' option

ok

>  I think we can split the space in the small samplers (next to Play) into two rows: top row shows just the title, while the bottom rows host BPM, Eject, xFade orientation (and maybe other controls)

done

> let's expand sampler rows like this (IMO all 8 samplers should be expanded, small samplers 5-8 are just for illustration):

I do not like the samplers expanded over 2 rows, I prefer the current solution, either the left 4 or the right for samplers can be expanded, not both.

Maybe you can add this 2x2 grid layout as an alternative expanding solution later on (as I have no idea how to get this layout done). 

The advantage on my expanding solution is that we somehow keep the skins similar. There might be a lot of users outside which like the current layout of their 4 samplers. Now they have the option to collapse them and use 8 samplers parallel.

=> I like to change the color of Sampler 5-8 to the same as deck 3/4

How can this be done easily?
adding color: #09B2AE; somewhere..

nopeppermint commented:
![8smallsampler](https://user-images.githubusercontent.com/3403218/47047731-a8add980-d198-11e8-8915-9b742a1e162e.png)
![1-4expanded](https://user-images.githubusercontent.com/3403218/47047734-aa779d00-d198-11e8-806d-b2f2dba06135.png)
![5-8expanded](https://user-images.githubusercontent.com/3403218/47047736-ac416080-d198-11e8-8cdb-1cb93a877f5a.png)

nopeppermint commented:
@ronso0 
any Idea how to get the "bpm" thing aligned left?

ronso0 commented:
Maybe it was to early to drop the 4/8 samplers option, because now right now the UX feels wrong:
- we have regular samplers 1-4 visible by default (empty config, clean install)
- only after collapsing samplers 1-4 it's revealed samplers 5-8 are also available
- the other 4 samplers (either 1-4 or 5-8) are inaccessible from the skin which is problematic/confusing especially if a) they are playing because they were activated just before pressing an expand button b) they are mapped to a controller and were loaded by the user, but now their on-screen counterpart is completely missing
- only after switching through samplers 1-4, samplers 5-8 and finally mini samplers 1-8 users can understand how it works in LateNight

I'm a bit torn apart how to proceed from here. I respect the case that there users familiar with the layout of 4 full-featured samplers who like to keep it that way. OTOH we need to consider new users checking out all skins to whom it should be obvious that we now have 8 samplers in LateNight as well.

This is my sugestion to keep the samplers layout consistent with the other skins as well as nicely accessible from skin and controllers:
- 8 samplers are the default
- small samplers are the default
- skin menu allows to switch back to 4 samplers
- all samplers are expanded at once
- 4 samplers expand to 1x4 (like before this PR)
- 8 samplers expand to 2x2 (1-4) and 2x2 (5-8) side by side (matches Shade skin where we also have 8 samplers in two 2x2 grids)

ronso0 commented:
> any Idea how to get the "bpm" thing aligned left?

Right now, the decks' Orientation, Keylock and Eject icons are used for the samplers. Those icons are defined inline (PushButton) and therefore are 18x25px to be equally spaced below the star rating.
In the samplers we need them to be compact, so I suggest to duplicate those icons and resize them to 17x18 or 18x18 and save them as sampler-specific versions (btn_keylock_sampler...) Then you'd just need to pack an expanding spacer (min,min) in between the BPM and fixed-sized sampler control icons. BPM is already aligned left; except the sampler is empty, then WNumber expands to available space and the "0.00" move to the right.

nopeppermint commented:
> matches Shade skin where we also have 8 samplers in two 2x2 grids
@ronso0 
in shade it's like that (correct me if I am wrong):
Samplers:
1234
5678

That's basically what we had in latenight but with two row's.


But what you and @Be-ing want is:
Samplers:
12   56
34   78

right?

This is expansion on the left (1-4) and expansion on the right (5-8) similar to latenight effect rack's

nopeppermint commented:
> 
    - all samplers are expanded at once

I overlooked this one, If there is only one expansion the realisation is much easier.
So the only open point for me is then the position of the samplers 1-8 as mention above.

nopeppermint commented:
> duplicate those icons, resize them

I agree

ronso0 commented:
> In shade it's like that (correct me if I am wrong)

I do ;)
In Shade it's actually
12  56
34  78
That's what matches most controllers where you'd have at least one row of 4 sample triggers **per side**

Be-ing commented:
> Samplers:
12 56
34 78

No,
1-2-3-4; 5-6-7-8
9-10-11-12; 13-14-15-16

ronso0 commented:
> > Samplers:
> > 12 56
> > 34 78
> 
> No,
> 1-2-3-4; 5-6-7-8
> 9-10-11-12; 13-14-15-16

Right now, we're talking about 8 samplers. So it's the same:
left side 1 2 3 4 // right side: 5 6 7 8
in expanded view, they're stacked exactly as mentioned (and as samplers are laid out in Shade)
> 1 2 | 5 6
> 3 4 | 7 8
and that works well with controllers with 4 sampler buttons in a row.

nopeppermint commented:
@ronso0 @Be-ing
The basic concept of 8 small samplers, with option to show only 4 samplers as backwards compatibility (4 sampler are always big, no small samplers if only 4 displayed) and option to expand them all 8 at once is now finished.

Is that what you had in mind?

ronso0 commented on res/skins/LateNight/samplers_unit_1-8.xml:
```diff
@@ -0,0 +1,96 @@
+          <ConfigKey>[Master],skin_settings</ConfigKey>
+          <BindProperty>visible</BindProperty>
+        </Connection>
+      </WidgetGroup>
```
This spacer is already in samplers_rack.xml, it can be removed here

nopeppermint replied:
ok, done

ronso0 commented on res/skins/LateNight/samplers_unit_1234.xml:
```diff
@@ -0,0 +1,46 @@
+            <ConfigKey>[Master],skin_settings</ConfigKey>
+            <BindProperty>visible</BindProperty>
+          </Connection>
+      </WidgetGroup>
```
This spacer is already in samplers_rack.xml, it can be removed here

nopeppermint replied:
ok, done

ronso0 commented on res/skins/LateNight/samplers_unit_1-8_big.xml:
```diff
@@ -0,0 +1,112 @@
+      <WidgetGroup><Size>2f,1min</Size></WidgetGroup>
+
+      <!-- Don't know why, but this button has to be inverted..-->
+      <PushButton>
```
please keep an eye on the indentations

nopeppermint replied:
ok, done

ronso0 commented on res/skins/LateNight/samplers_unit_1-8_big.xml:
```diff
@@ -0,0 +1,112 @@
+
+        <!-- First row -->
+        <WidgetGroup>
+        <ObjectName>SamplerTextSmall</ObjectName>
```
here & below: please keep an eye on the indentations

nopeppermint replied:
ok, done

ronso0 commented on res/skins/LateNight/samplers_unit_1-8_big.xml:
```diff
@@ -0,0 +1,112 @@
+          <ConfigKey>[Master],skin_settings</ConfigKey>
+          <BindProperty>visible</BindProperty>
+        </Connection>
+      </WidgetGroup>
```
This spacer is already in samplers_rack.xml, it can be removed here

nopeppermint replied:
ok, done

ronso0 commented on res/skins/LateNight/skin.xml:
```diff
@@ -39,9 +39,12 @@
       <attribute persist="true" config_key="[Microphone],show_microphone">0</attribute>
       <attribute persist="true" config_key="[EffectRack1],show">0</attribute>
       <attribute persist="true" config_key="[Samplers],show_samplers">0</attribute>
+      <attribute persist="true" config_key="[Master],show_4sampler">0</attribute>
```
By "8 samplers are default" I meant to use a CO like [Skin],8_samplers and set it to 1.
This is consistent with [..],show_4deck, [..],show_4effectunits & [..],8_hotcues

nopeppermint replied:
ok, done

ronso0 commented on res/skins/LateNight/skin_settings.xml:
```diff
@@ -263,30 +263,41 @@ Description:
+                  </WidgetGroup>
+                  <Template src="skin:skin_settings_button_2state.xml">
+                    <SetVariable name="text">only 4 Samplers</SetVariable>
+                    <SetVariable name="Setting">[Master],show_4sampler</SetVariable>
```
as mentioned above: please use "8 Samplers" and  [..],8_samplers

nopeppermint replied:
ok, done

ronso0 commented on res/skins/LateNight/samplers_rack.xml:
```diff
@@ -1,43 +1,68 @@
-        <SetVariable name="samplernum">3</SetVariable>
-      </Template>
+             <!-- Select between 5-8 big or 8 tiny -->
+                  <WidgetStack currentpage="[Samplers],show_big">
```
[Skin],expand_samplers

ronso0 replied:
please keep an eye on the indentations, and please make the comments match the actual widgets

ronso0 commented on res/skins/LateNight/samplers_rack.xml:
```diff
@@ -1,43 +1,68 @@
 
-      <WidgetGroup><Size>10f,1min</Size></WidgetGroup>
+                      <!--8 tiny Sampler -->
+                      <WidgetGroup trigger="[Samplers],show_big5678_big_of_8" on_hide_select="0">
```
should be the same as above: [Skin],expand_samplers

nopeppermint replied:
ok, done

ronso0 commented on res/skins/LateNight/samplers_unit_1-8_big.xml:
```diff
@@ -0,0 +1,112 @@
+            <ConfigKey>[Samplers],show_big</ConfigKey>
+            <ButtonState>LeftButton</ButtonState>
+          </Connection>
+      </PushButton>
```
IMO this button should be at the far right, like the FX expand buttons

ronso0 commented:
> I think it's confusing to break what was 8 samplers on one row into multiple rows.

I think the grouping will be clear if we use a significant spacer in the center of the collapsed samplers, as well.

nopeppermint commented:
@ronso0 thanks for the review, I will look into your points.

nopeppermint commented on res/skins/LateNight/samplers_rack.xml:
```diff
@@ -1,43 +1,68 @@
-        <SetVariable name="samplernum">3</SetVariable>
-      </Template>
+             <!-- Select between 5-8 big or 8 tiny -->
+                  <WidgetStack currentpage="[Samplers],show_big">
```
thanks

ronso0 commented:
I hope I don't request too much.. (thanks for your work so far :)
Big samplers 1-4 are used in two templates (so they are built twice) therefore they could be instantiated as Singletons in samplers_rack.xml

Also, please order the samplers like below and add a wider spacer in between small samplers 4 & 5, so we can check if the concept works.
1 2 5 6
3 4 7 8

nopeppermint commented:
> IMO this button should be at the far right, like the FX expand buttons

> I think it's confusing to break what was 8 samplers on one row into multiple rows.

>I think the grouping will be clear if we use a significant spacer in the center of the collapsed samplers, as well.

what do you think about the button in the middle?
![small](https://user-images.githubusercontent.com/3403218/47322245-97147800-d657-11e8-9dd8-87b7e194e7fe.png)
![big](https://user-images.githubusercontent.com/3403218/47322246-9845a500-d657-11e8-9413-3e0b2d39e25e.png)

nopeppermint commented:
> Also, please order the samplers like below and add a wider spacer in between small samplers 4 & 5, so we can check if the concept works.
1 2 5 6
3 4 7 8

![](https://user-images.githubusercontent.com/3403218/47322245-97147800-d657-11e8-9dd8-87b7e194e7fe.png)


![big2](https://user-images.githubusercontent.com/3403218/47322397-0db17580-d658-11e8-8663-190bdf9ef55b.png)

ronso0 commented:
:+1:
the expand button should stick to the top, and maybe it would also work with less space around it.
The small samplers are taller than [earlier](https://user-images.githubusercontent.com/3403218/47047731-a8add980-d198-11e8-8915-9b742a1e162e.png)
I'll send a fix soon.

ronso0 committed:
refactor layout, fix expand button, add 4 small samplers

ronso0 commented:
0a1a280efa7a47901cc5764ce7d49da67c7ab2c3
this also allows to collapse 4 samplers. I think we should allow this option there, as well.

ronso0 commented:
![latenight-4-small](https://user-images.githubusercontent.com/5934199/47327711-a6eb8680-d66e-11e8-8c9b-e339892a0680.png)
![latenight-4-big](https://user-images.githubusercontent.com/5934199/47327712-a81cb380-d66e-11e8-9a8d-10cab311e94c.png)

ronso0 commented:
I'm not 100% happy with the wrench icon, yet, but I think it's consistent with the beatgrid expand icon, as it allows expands sampler 'tools'.
+/- is used in Shade for all expand toggles, but in LateNight its only occurence would be on the sampler expand button, furthermore I'd expect it to add/remove elements instead of unrolling elements.
Are you okay with that?

ronso0 commented:
I notice the BPM info is only available in small samplers. Though this is not consistent across skins, should we put next to the title in big samplers?

nopeppermint committed:
change samplerexpand button logo, add bpm to big sampler

nopeppermint commented:
> I'm not 100% happy with the wrench icon, yet,

I prefer another Icon as well, what do you think about
![btn_samplercollapse svg](https://user-images.githubusercontent.com/3403218/47379945-997cdd80-d6fc-11e8-8221-5bc688409ea3.png)

> I notice the BPM info is only available in small samplers. Though this is not consistent across skins, should we put next to the title in big samplers?

jep, done

nopeppermint commented:
![expanded](https://user-images.githubusercontent.com/3403218/47383310-c208d580-d704-11e8-9c1a-cba4df28982b.png)
![collected](https://user-images.githubusercontent.com/3403218/47383376-dcdb4a00-d704-11e8-8699-097d6976a703.png)


what do you think about that?

daschuer commented:
both, ... and hamburger menu have a different meaning. 
For me most natural is a turnning '>' or '<- ->'  /  '-><-'

I choose +/- in shade, because it is the tree view expand and collapse icon in Win98

daschuer commented:
looks like a fast forward button ....

nopeppermint commented:
> looks like a fast forward button ....

;-)

I like the idea of  '<- ->' / '-><-'

ronso0 commented:
+- makes total sense in Shade98 ;)

So let's adopt the arrow from Tango?
![btn_samplerexpand_4](https://user-images.githubusercontent.com/5934199/47383715-95a18900-d705-11e8-8f31-82d7c8411c65.png)

ronso0 commented:
'<- ->' / '-><-' works well in Deere because, there, buttons don't have borders

nopeppermint commented:
![collapsed](https://user-images.githubusercontent.com/3403218/47384130-87a03800-d706-11e8-8d0a-2c4e58ce1e2c.png)
![expanded](https://user-images.githubusercontent.com/3403218/47384134-8969fb80-d706-11e8-8fe1-33dd919c0369.png)

I like those

ronso0 commented:
If we find a nice icon that's small enough to fit the FX expand button, that would be great. That's because I doubt the FX expand buttons are intuitive right now, it's just the border gradient that somehow indicates a special function.

nopeppermint commented:
I would chose '<- ->' / '-><-'  for samplers and let the fx as they are.
But ok, let's give creative icon designers some time to join this PR if they not already are  ;-)

ronso0 committed:
tune expand/collapse icons

LateNight: optimize SVGs, decrease Eject & Keylock icons

nopeppermint commented:
thanks @ronso0 
ready to merge

ronso0 commented:
final touch: decrease samplers' Eject & Keylock icons, optimize SVGs
Samplers don't expand anymore when tracks with three-digit BPM are loaded.
56d5a2458c

![latenight-8samplers-final](https://user-images.githubusercontent.com/5934199/47505182-f1872180-d86d-11e8-8432-9b717d067bf1.png)

nopeppermint commented:
> final touch: decrease samplers' Eject & Keylock icons, optimize SVGs
Samplers don't expand anymore when tracks with three-digit BPM are loaded.
56d5a24

thanks

Be-ing commented:
I still think the rearrangement when expanding is a bit confusing, but now the samplers have been reordered it is not as bad.

Use the button templates like in #1858 or manually specify the `scalemode="STRETCH_ASPECT"` attribute for all SVG paths to ensure everything looks smooth on high pixel density screens.

When I load samplers, shut down with the samplers collapsed, and restart Mixxx, when I expand the samplers, only some of the overview waveforms are drawn:
![screenshot from 2018-10-25 13-56-03](https://user-images.githubusercontent.com/9455094/47523482-c563a480-d85d-11e8-99bf-7260fb2a68b3.png)
If I load new tracks to the samplers, reload the skin in Developer Mode, or shut down Mixxx with the samplers expanded, this does not happen.

At first I noticed a bug where the sampler play buttons would only stay lit if I clicked and held them, but not if I short pressed them. However, I can't reproduce that now.

nopeppermint commented:
> When I load samplers, shut down with the samplers collapsed, and restart Mixxx, when I expand the samplers, only some of the overview waveforms are drawn:

I can reproduce this error.
But if you do not collapse them before closing, you could see the same effect on starting, but only for a short period of time, it then displays:
![startup](https://user-images.githubusercontent.com/3403218/47530101-f36ed600-d8a9-11e8-8bde-337479b275ef.png)

This might be a hint for solving this Bug.

> Use the button templates like in #1858 or manually specify the scalemode="STRETCH_ASPECT"

I do not have a 4k monitor, therefore I am not able to test this.
Furthermore  I do not know if `scalemode="STRETCH_ASPECT"` is needed for EACH SVG as in #1858 not all SVG have `scalemode="STRETCH_ASPECT"`  added...

maybe you could fix this in another PR?

ronso0 commented:
Neither can I confirm missing waveforms, nor do I see the anylyzing text. Re-analyzing is done around when the skin is fully loaded.

ronso0 commented:
I'll send a fix that migrates to button templates...has been on my LateNight list anyway

Be-ing committed:
LateNight: use button templates in samplers

fixes scaling with high pixel density screens (the templates use
scalemode="STRETCH_ASPECT"). I do not know why setting
scalemode="STRETCH_ASPECT" for the crossfader assignment and
sampler expansion <Unpressed> elements does not work.

nopeppermint commented:
reabased

nopeppermint commented:
I think I found the problem for this:

> When I load samplers, shut down with the samplers collapsed, and restart Mixxx, when I expand the samplers, only some of the overview waveforms are drawn:

![](https://user-images.githubusercontent.com/9455094/47523482-c563a480-d85d-11e8-99bf-7260fb2a68b3.png)

The sampler expansion is not new, also skin deere use it, but there we do not have two file for samplers (smallsampler and sampler). Therefore the waveform is always drawn on startup but with not expanded samplers just not visible. In latenight we started we two files for the samplers, so if we do as @Be-ing mentioned above the waveform will not be drawn on restart as we have two different samplers file..

So the solution might be to merge smallsampler.xml with samperl.xml and put
```    
<!-- Select between big or small samplers -->
<WidgetStack currentpage="[Skin],expand_samplers">
```

inside there as well.

Or pack `<Overview>` into a singleton and draw it in smallsampler.xml (not visible) and sampler.xml (visible).

I already tried to simply but `<Overview>` as well into smallsampler.xml but then the same issue happens as then there are two `<Overview>` per sampler, one from the smallsampler.xml and  one from sampler.xml

Other (maybe easier) Idea, modify cpp code to trigger redraw of sampler (even if they are not shown) on startup of mixxx.

@Be-ing @ronso0  what do you think?

ronso0 commented:
@Be-ing Do you notice the issue in Shade as well?
There we also have separate templates for small and big samplers.

I rather think the WidgetStack might be the issue. I reverted to `WidgetGroups` with `<Connection>` with 9020f1d9c9

Be-ing commented:
> @Be-ing Do you notice the issue in Shade as well?

Yes

> I rather think the WidgetStack might be the issue. I reverted to WidgetGroups with <Connection> with 9020f1d

This did not fix the issue for me.

ronso0 commented:
> This did not fix the issue for me.

Okay. Does it work if you merge e950807eee ? This also removes the widgetstack from the sampler rack

ronso0 commented:
> Does it work if you merge e950807 

.. top of 9020f1d9c9114932646bc0601b1a223c436542f6 ?

ronso0 commented:
...and if that doesn't help, please put 6e574140d8 on top, as well.

I'm curious why I don't have any issues at all. Might it related to the waveform render?

Be-ing commented:
None of those fixed the bug, neither did switching from RGB (GL) to RGB waveform renderers.

nopeppermint commented:
> @Be-ing Do you notice the issue in Shade as well?

It happens as well on my site with Shade

I am using Linux Mint 19 x64

nopeppermint commented:
I can confirm that:
`9020f1d` does not help
`e950807` does not help
`6e57414`does not help

ronso0 commented:
> Neither can I confirm missing waveforms, nor do I see the anylyzing text. Re-analyzing is done around when the skin is fully loaded.

I can see the analyzing text, too, didn't notice ealier because it blended with the waveforms.

One strange thing I noticed is the samples are re-analyzed in random order:
 3 > 5 > 6 > 1 > [~5sec of processing invisble samplers 9-64?] > 7 
@Be-ing @nopeppermint How many samplers do you have loaded? Is there a difference if you load only 8 samplers? (test with Deere 64, or export the sample bank and check the file)

Be-ing commented:
I had 8 samplers loaded. I also noticed sometimes some of the overview waveforms were loaded on startup, but it was random which ones those were.

nopeppermint commented:
can we merge this to 2.2 and fix this minor bug later?
I mean it's not a new bug, it already there with Skin Shade.

nopeppermint commented:
rebased

ronso0 commented:
> can we merge this to 2.2 and fix this minor bug later?
I mean it's not a new bug, it already there with Skin Shade.

I agree. Did you file a bug?

nopeppermint commented:
we could also change in skin.xml:
from "true" to false and "0" to "1" 
<attribute persist="false" config_key="[Skin],expand_samplers">1</attribute>
then it would always open the samplers big, you then have the problem only if you decide to completely hide the samplers..

but this is not really fixing anything..

ronso0 commented:
> rebased

when you use `git rebase -i remote/branch` (interactive rebase) you can also squash commits = combine commits, like `change samplerexpand button logo, add bpm to big sampler` + `change expand button` + `change expand button` = `change expand icon, add BPM to big sampler`
This clears the commit history a bit..

nopeppermint commented:
> I agree. Did you file a bug?

https://bugs.launchpad.net/mixxx/+bug/1801126

Be-ing commented:
Please use more descriptive commit messages than "fix". You can edit commit messages with `git rebase -i`

nopeppermint commented:
anything else? @ronso0  @Be-ing

nopeppermint commented:
done, now we are down on 5 commits (from about 25 ...)

rryan commented:
thanks @nopeppermint !
